Mathlib/Algebra/Order/GroupWithZero/Unbundled.lean

```diff
@@ -1512,6 +1515,10 @@ lemma zpow_right_injective₀ (ha₀ : 0 < a) (ha₁ : a ≠ 1) : Injective fun
 @[simp] lemma zpow_right_inj₀ (ha₀ : 0 < a) (ha₁ : a ≠ 1) : a ^ m = a ^ n ↔ m = n :=
   (zpow_right_injective₀ ha₀ ha₁).eq_iff
 
+lemma eq_zero_of_zpow_eq_one₀ (ha₀: 0 < a) (ha₁ : a ≠ 1) {n : ℤ}: a ^ n = 1 ↔ n = 0 := by
```

Comment:
```suggestion
lemma zpow_eq_one₀ (ha₀: 0 <= a) (ha₁ : a ≠ 1) {n : ℤ} : a ^ n = 1 ↔ n = 0 := by
```
and can you please remove the `a = 0` special case you are introducing in this PR?

Sorry, writing from phone

Reply:
I don't understand. I am assuming `0 < a` in the lemma.

Reply:
Yes, but it's also true for `a = 0` (as per your lemma above), so you can generalise the proof 😉

Reply:
ah that's what you meant. ok, I've generalized it.